action.yml

```diff
@@ -13,22 +13,47 @@ inputs:
   precompile:
     description: 'Whether to allow auto-precompilation (via the `JULIA_PKG_PRECOMPILE_AUTO` env var). Options: yes | no. Default value: no.'
     default: 'no'
+  propagate_registry_preference:
+    description: 'Propagate the JULIA_PKG_SERVER_REGISTRY_PREFERENCE environment variable to future steps'
+    default: 'yes'
 
 runs:
   using: 'composite'
   steps:
-    # Occasionally, there are rather large delays (> a few hours)
-    # between the time a package is registered in General and
-    # propagated to pkg.julialang.org.  We can avoid this by manually
-    # cloning ~/.julia/registries/General/ in Julia 1.5 and later.
-    # See:
-    # * https://github.com/JuliaLang/Pkg.jl/issues/2011
-    # * https://github.com/JuliaRegistries/General/issues/16777
-    # * https://github.com/JuliaPackaging/PkgServer.jl/issues/60
-  - run: julia --color=yes "$GITHUB_ACTION_PATH"/add_general_registry.buildpkg.jl
-    shell: bash
-
-  - run: julia --color=yes --project=${{ inputs.project }} -e 'using Pkg; if VERSION >= v"1.1.0-rc1"; Pkg.build(verbose=true); else Pkg.build(); end'
-    shell: bash
+  - run: |
+      import Pkg
+      pref_key = "JULIA_PKG_SERVER_REGISTRY_PREFERENCE"
+      pref_value = strip(get(ENV, pref_key, "eager"))
+      ENV[pref_key] = pref_value
+      input = strip(ENV["PROPAGATE_INPUT"])
```

Comment:
What is this `PROPAGATE_INPUT` stuff? I'm confused why this fix is not a lot simpler; just set the env variable and done? It seems it is adding these extra options that are pretty unlikely to be used in practice. If someone wants some special behavior they can just add that to their own CI files.

Reply:
I was about to say the same. If you need this then it should go in the root environment key of the CI job description. It seems strange that it is "hidden" in the buildpkg action (which most packages should not need to use anyway).